.github/workflows/release.yml

```diff
@@ -196,7 +205,21 @@ jobs:
             GEOS_INCLUDE_PATH='${{ runner.temp }}\geos-${{ env.GEOS_VERSION }}\include'
           CIBW_BEFORE_ALL_MACOS: ./ci/install_geos.sh
           CIBW_BEFORE_ALL_WINDOWS: ci\install_geos.cmd
-          CIBW_BEFORE_BUILD_WINDOWS: pip install delvewheel
+          # TEMP don't use automated/isolated build environment, but manually
+          # install build dependencies so we can build with cython 3.1.0a0
+          # once Cython and numpy have official releases that support free threading we can remove this
+          CIBW_BUILD_FRONTEND: "pip; args: --no-build-isolation"
+          CIBW_BEFORE_BUILD: |
+            # For free threading support in Cython install a pre-release version of Cython
+            PRE_FLAG="$(python -c"import sysconfig; print('--pre' if sysconfig.get_config_var('Py_GIL_DISABLED') else '')")" &&
+            pip install $PRE_FLAG cython &&
+            pip install numpy setuptools wheel
+          CIBW_BEFORE_BUILD_WINDOWS:
+            # For free threading support in Cython install a pre-release version of Cython
+            PRE_FLAG="$(python -c"import sysconfig; print('--pre' if sysconfig.get_config_var('Py_GIL_DISABLED') else '')")" &&
+            pip install $PRE_FLAG cython &&
```

Comment:
This is problematic in powershell apparently. I tried to figure this out briefly on my own fork, but couldn't get the right Windows incantation here with several different attempts. Does anyone else have Windows knowledge or know what I need to do here to get this to work under Windows?

```
PRE_FLAG="$(python -c"import sysconfig; print('--pre' if sysconfig.get_config_var('Py_GIL_DISABLED') else '')")" && pip install $PRE_FLAG cython && pip install numpy setuptools wheel && pip install delvewheel
  'PRE_FLAG' is not recognized as an internal or external command,
  operable program or batch file.
```
https://github.com/shapely/shapely/actions/runs/11945885851/job/33427753116?pr=2138#step:7:458

Reply:
I'm terrible at the various Windows shells, so I'd change to `bash` or try to steal the invocation from another project. E.g., from NumPy:

https://github.com/numpy/numpy/blob/59647e846b6c25f0e44905936693d9a49c1e0ac3/tools/wheels/cibw_before_build.sh#L59

Reply:
Or with an if/else instead of trying to pass the flag to the single pip call?

Reply:
I could not get the Windows shell to cooperate. I fell back to an if/else within a Python subprocess to call the respective pip install command instead (also used this in the linux BEFORE_BUILD steps for consistency).

Reply:
Given this is only temporary until cython does a release, let's just go with this since it is working!